### ISSUE TITLE
Non-required integer parameter becomes required in Swagger UI 3.x

### ISSUE THREAD

**Issue by frol** [swagger-api/swagger-ui#3361]
Swagger version: 3.0.18 (2.x worked just fine)

Swagger YAML:

```yaml
swagger: "2.0"
info:
  title: "title"
  version: "1.0.0"
paths:
  /pets/:
    get:
      parameters:
      - name: "limit"
        in: "query"
        required: false
        type: "integer"
      responses:
        200:
          description: "successful operation"
```

"Try it out" blocks the request submission and highlights the `limit` field with `invalid` CSS class on "Execute".

**Comment by emtiazahmed** [comment]
Hi guys,
Any ETA on this or can someone point me to an earlier version that used to work?

Thanks

**Comment by kliyes** [comment]
I'm facing the same issue and already get back to v3.0.17. There's already a PR should fix this.

**Comment by kliyes** [comment]
It is **not** fixed in 3.0.19, did it?
![image](https://user-images.githubusercontent.com/3313588/28354494-a9cbdad6-6c92-11e7-9457-592c9e69f9c1.png)

**Comment by frol** [comment]
@kliyes 3.0.19 works fine for me.
